os/board/rtl8721csm/src/component/common/api/wifi/wifi_conf.c

```diff
@@ -2827,4 +2831,119 @@ int wifi_set_null1_param(uint8_t check_period, uint8_t limit, uint8_t interval)
 }
 #endif
 
+#include "freertos/wrapper.h"
+extern Rltk_wlan_t rltk_wlan_info[NET_IF_NUM];
+unsigned char wifi_monitor_on = 0;
+static pthread_t wifimoninfo;
+
+typedef struct WIFI_MONITOR{
+	/*RX related*/
+	unsigned long rx_success_pkt_count;
+	unsigned long rx_success_bytes_count;
+	unsigned long rx_dropped_pkt_count;
+	int rx_rssi;
+	u32 rxd_beacon_count;
+
+	/*TX related*/
+	unsigned long tx_success_pkt_count;
+	unsigned long tx_success_bytes_count;
+	unsigned long tx_dropped_pkt_count;
+}wifi_monitor;
+
+static wifi_monitor wifi_monitor_info = {0};
+static wifi_monitor wifi_monitor_prev = {0};
+
+void wifi_get_monitor_info()
+{
+	struct net_device *dev = NULL;
+	struct net_device_stats *stats;
+	int rssi = 0;
+	int idx = 0;
+	u32 total_beacon_count = 0;
+	u32 rxd_beacon_count = 0;
+	u32 tmp_rxd_beacon_count = 0;
+	u32 missed_beacon_count = 0;
+
+	save_and_cli();
+
+	if(rltk_wlan_info[idx].enable) {
+		/*to get TX & RX packets status*/
+		dev = rltk_wlan_info[idx].dev;
+		stats = dev->get_stats(dev);
+		wifi_monitor_info.rx_success_pkt_count = stats->rx_packets;

```

Comment:
Same as above about `stats`.

Reply:
If the condition "rltk_wlan_info[idx].enable" is true, there is no need to check "dev" and "stats" separately.